regparser/tree/xml_parser/paragraph_processor.py

```diff
@@ -61,7 +61,10 @@ def select_depth(self, depths):
         for fn, weight in self.DEPTH_HEURISTICS.items():
             depths = fn(depths, weight)
         depths = sorted(depths, key=lambda d: d.weight, reverse=True)
-        return depths[0]
+        if depths:
+            return depths[0]
+        else:
+            return depths
```

Comment:
This is confusing, if depths is empty this will return an array.. an error here provides clearer intent of the function.